src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1701,7 +1701,7 @@ struct sd_bus_vtable_original {
 #define VTABLE_ELEMENT_SIZE sizeof(struct sd_bus_vtable)
 
 static int vtable_features(const sd_bus_vtable *vtable) {
-        if (vtable[0].x.start.element_size == VTABLE_ELEMENT_SIZE_ORIGINAL ||
+        if (vtable[0].x.start.element_size < VTABLE_ELEMENT_SIZE ||
             !vtable[0].x.start.vtable_format_reference)
                 return 0;
         return vtable[0].x.start.features;
```

Comment:
I don't understand this change. Imagine a future modification to `struct sd_bus1_vtable`, which increases its size and as such `VTABLE_ELEMENT_SIZE` will increase as well. If you now run an old binary against this newly compiled library, `vtable_features()` will return 0, even though the old binary might have actually put features into the structure.

Or am I reading this wrong? Is the intention to not update `VTABLE_ELEMENT_SIZE` here when it changes again?

Reply:
Yes, the idea is to update the check to use the current (as of now) size in the future. I renamed the macros and added a comment to make this clear.